src/withForm.js

```diff
@@ -9,37 +9,42 @@ const withForm = (BaseComponent) => {
     }
     setInitialValues = (initialValues, currentValues) => {
       initialValues = fromJS(initialValues)
-      this.setState({
+      this.setState((state) => ({
+        ...state,
         initialValues,
         currentValues: currentValues ? fromJS(currentValues) : initialValues
-      })
+      }))
     }
     setValue = (field, value) => {
-      this.setState({
+      this.setState((state) =>  ({
+        ...state,
         currentValues: this.state.currentValues.set(field, fromJS(value))
```

Comment:
should this be `state.` instead of `this.state.`?

Reply:
good catch